src/main/java/carpet/CarpetSettings.java

```diff
@@ -739,4 +741,47 @@ public static void changeSpawnSize(int size)
             category = {SURVIVAL, CLIENT}
     )
     public static boolean cleanLogs = false;
+
+    @Rule(
+            desc = "Customizable structure block limit of each axis",
+            extra = {"WARNING: Needs to be permanent for correct loading.",
+                    "Setting 'structureBlockIgnored' to air is recommended",
+                    "when saving massive structures.",
+                    "Required on client of player editing the Structure Block.",
+                    "'structureBlockOutlineDistance' may be required for",
+                    "correct rendering of long structures."},
+            options = {"48", "96", "192", "256"},
+            category = {CREATIVE, CLIENT},
+            strict = false
+    )
+    public static int structureBlockLimit = 48;
+
+    public static class StructureBlockIgnoredValidator extends Validator<String> {
+
+        @Override
+        public String validate(ServerCommandSource source, ParsedRule<String> currentRule, String newValue, String string) {
+            if (!Registry.BLOCK.getIds().contains(Identifier.tryParse(newValue))) {
+                Messenger.m(source, "Unknown block '" + newValue + "'.");
+                return null;
+            }
+            return newValue;
+        }
+    }
+    @Rule(
+            desc = "Changes the block ignored by the Structure Block",
+            options = {"minecraft:structure_void", "minecraft:air"},
+            category = {CREATIVE},
+            validate = StructureBlockIgnoredValidator.class
+    )
+    public static String structureBlockIgnored = "minecraft:structure_void";
+
+    @Rule(
+            desc = "Customizable Structure Block outline render distance",
+            extra = "Required on client to work properly",
+            options = {"48", "96", "192"}, // 48 is the default structure block limit, and 192 is 12 chunks
+            category = {CREATIVE, CLIENT},
+            strict = false,
+            validate = Validator.NONNEGATIVE_NUMBER.class
+    )
+    public static double structureBlockOutlineDistance = 96d;
```

Comment:
again like with the reach limits - not sure why you would like to suggest the option to shorten that distance if vanilla is 96. I think the ony usecase is to extend that. 

Since its purely cosmetic value I woudn't be limiting it to >=96 here since using 0 may also have some usecases (I don't want to see these lines), but NONNEGATIVE_NUMBER (from built-in validators) can be used here.

Reply:
Already used the `NONNEGATIVE_NUMBER` validator there.  
I will remove the "46" value proposal.